src/generic/Matrix.jl

```diff
@@ -28,32 +28,40 @@ export MatrixSpace, fflu!, fflu, solve_triu, isrref, charpoly_danilevsky!,
 #
 ###############################################################################
 
-function similar(x::Mat{T}) where T <: RingElement
-   R = base_ring(x)
-   M = similar(x.entries)
+@doc Markdown.doc"""
```

Comment:
There's a closing parenthesis missing on the first prototype. Also the docstrings need to be referred to in matrix.md in the docs directory.

Reply:
Done (in matrix_spaces.md`, as there was no reference to `similar` in `matrix.md`), also I added an implementation for `MatAlgElem`.